src/ralph/dns/dnsaas.py

```diff
@@ -0,0 +1,178 @@
+from ralph.dns.forms import RecordType
+from ralph.helpers import cache
+
+logger = logging.getLogger(__name__)
+
+
+class DNSaaS:
+
+    def __init__(self, headers=None):
+        self.session = requests.Session()
+        _headers = {
+            'Authorization': 'Token {}'.format(settings.DNSAAS_TOKEN)
+        }
+        if headers is not None:
+            _headers.update(headers)
+        self.session.headers.update(_headers)
+
+    def get_api_result(self, url):
+        """
+        Returns 'results' from DNSAAS API.
+
+        Args:
+            url: Url to API
+
+        Returns:
+            list of records
+        """
+        request = self.session.get(url)
+        json_data = request.json()
+        api_results = json_data.get('results', [])
+        if json_data.get('next', None):
+            _api_results = self.get_api_result(json_data['next'])
+            api_results.extend(_api_results)
+        return api_results
+
+    def get_dns_records(self, ipaddresses):
+        """Gets DNS Records for `ipaddresses` by API call"""
+        dns_records = []
+        ipaddresses = [('ip', i) for i in ipaddresses]
+        url = urljoin(
+            settings.DNSAAS_URL,
+            'api/records/?{}'.format(
+                urlencode([
+                    ('limit', 100),
+                    ('offset', 0)
+                ] + ipaddresses)
+            )
+        )
+        api_results = self.get_api_result(url)
+        ptr_list = set(
+            [i['content'] for i in api_results if i['type'] == 'PTR'])
+
+        for item in api_results:
+            if item['type'] in {'A', 'CNAME', 'TXT'}:
+                dns_records.append({
+                    'pk': item['id'],
+                    'name': item['name'],
+                    'type': RecordType.from_name(item['type'].lower()).id,
+                    'content': item['content'],
+                    'ptr': item['name'] in ptr_list and item['type'] == 'A',
+                    'owner': settings.DNSAAS_OWNER
+                })
+        return dns_records
+
+    def update_dns_records(self, record):
+        """
+        Update DNS Record in DNSAAS
+
+        Args:
+            record: record cleaned data
+
+        Returns:
+            True or False if an error from api
+        """
+        url = urljoin(
+            settings.DNSAAS_URL, 'api/records/{}/'.format(record['pk'])
+        )
+        data = {
+            'name': record['name'],
+            'type': RecordType.raw_from_id(int(record['type'])),
+            'content': record['content'],
+            'auto_ptr': (
+                settings.DNSAAS_AUTO_PTR_ALWAYS if record['ptr'] and
+                record['type'] == str(RecordType.a.id)
+                else settings.DNSAAS_AUTO_PTR_NEVER
+            ),
+            'owner': settings.DNSAAS_OWNER
+        }
+        request = self.session.patch(url, data=data)
+        if request.status_code != 200:
+            return False
+        return True
+
+    @cache()
```

Comment:
this cache here is useless since almost all record will be different. The only cache which makes sense here is caching based on `domain_name` param.